validatorobserver/validator_observer.go

```diff
@@ -0,0 +1,197 @@
+package validatorobserver
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/Ethernal-Tech/apex-bridge/common"
+	"github.com/Ethernal-Tech/apex-bridge/contractbinding"
+	"github.com/Ethernal-Tech/apex-bridge/eth"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/hashicorp/go-hclog"
+)
+
+type ValidatorSetObserver struct {
+	validatorSetStream  chan *Validators
+	validators          Validators
+	validatorSetPending bool
+	bridgeSmartContract eth.IBridgeSmartContract
+	logger              hclog.Logger
+}
+
+type ValidatorsChainData struct {
+	Keys       []eth.ValidatorChainData
+	SlotNumber uint64
```

Comment:
this slot should be updated in execute function also. for each chain if we are in pending validator set update

Reply:
this is not one time data. it is data freezed by ab after validator set update starts